src/awscr-s3/file_uploader.cr

```diff
@@ -16,6 +26,8 @@ module Awscr::S3
     # uploader.upload("bucket1", "obj", IO::Memory.new("DATA!"))
     # ```
     def upload(bucket : String, object : String, io : IO, headers : Hash(String, String) = Hash(String, String).new)
+      headers = @options.with_content_types ? headers.merge!(content_type_header(io)) : headers
```

Comment:
What do you think about using the non bang merge, i know we are in other places but it seems reasonable to not modify input

Reply:
Yes I agree. Only used it because it was used other places lol. I'll change.